org_fedora_oscap/scap_content_handler.py

```diff
@@ -0,0 +1,205 @@
+
+    def get_data_streams_checklists(self):
+        """
+        Method to get data streams and their checklists found in the SCAP
+        source data stream represented by the SCAPContentHandler.
+
+        :return: a dictionary consisting of the IDs of the data streams as keys
+                 and lists of their checklists' IDs as values
+                 None if the file isn't a SCAP source data stream
+        :rtype: dict(str -> list of strings)
+        """
+        if self.scap_type != "SCAP_SOURCE_DATA_STREAM":
+            return None
+        checklists = {}
+        for data_stream in self.root.findall("ds:data-stream", ns):
+            data_stream_id = data_stream.get("id")
+            crefs = []
+            for cref in data_stream.findall(
+                    "ds:checklists/ds:component-ref", ns):
+                cref_id = cref.get("id")
+                crefs.append(cref_id)
+            checklists[data_stream_id] = crefs
+        return checklists
+
+    def _parse_profiles_from_xccdf(self, benchmark):
+        if benchmark is None:
+            return []
+        if benchmark.tag.startswith(f"{{{ns['xccdf-1.1']}}}"):
+            xccdf_ns_prefix = "xccdf-1.1"
+        elif benchmark.tag.startswith(f"{{{ns['xccdf-1.2']}}}"):
+            xccdf_ns_prefix = "xccdf-1.2"
+        else:
+            raise SCAPContentHandlerError("Unsupported XML namespace")
+        profiles = []
+        for profile in benchmark.findall(f"{xccdf_ns_prefix}:Profile", ns):
+            profile_id = profile.get("id")
+            title = profile.find(f"{xccdf_ns_prefix}:title", ns)
+            description = profile.find(f"{xccdf_ns_prefix}:description", ns)
+            if description is None:
+                description_text = ""
+            else:
+                description_text = parse_HTML_from_content(description.text)
+            profile_info = ProfileInfo(
+                profile_id, title.text, description_text)
+            profiles.append(profile_info)
+        # if there are no profiles we would like to prevent empty profile
+        # selection list in the GUI so we create the default profile
+        if len(profiles) == 0:
+            default_profile = ProfileInfo(
+                "default",
+                "Default",
+                "The implicit XCCDF profile. Usually, the default profile "
+                "contains no rules.")
+            profiles.append(default_profile)
+        return profiles
+
+    def select_checklist(self, data_stream_id, checklist_id):
+        """
+        Method to select a specific XCCDF Benchmark using
+        :param data_stream_id: value of ds:data-stream/@id
+        :type data_stream_id: str
+        :param checklist_id: value of ds:component-ref/@id pointing to
+            an xccdf:Benchmark
+        :type checklist_id: str
+        :return: None
+
+        """
+        self._data_stream_id = data_stream_id
+        self._checklist_id = checklist_id
+
+    def get_profiles(self):
```

Comment:
Those nested ifs are confusing. I propose to restructure it in the following way:

Firs identify all pathological cases, throw all the easy errors - bad XCCDF, standalone tailoring or unknown content.
Then, there is the trivial benchmark extraction from XCCDF, and multi-step benchmark extraction from a datastream, that I propose to extract to another method.